### ISSUE TITLE
Proposal: Ability not to auto-close comments

### ISSUE THREAD

**Issue by erichiller** [microsoft/vscode#1966]
As far as I can tell this is not currently available. 

Oftentimes I need to comment out blocks of code and use `/**` and immediately a `*/` auto inserted resulting in one extra step.  Not a huge deal by any means, but it would be a nice added feature to be able to turn off auto-closures on comments. Or maybe even in the configs an array of chars not to close? If nobody else feels this way though, probably not worth the effort. I suppose I am looking for input on this feature.

**Comment by isidorn** [comment]
We currently have `editor.autoClosingBrackets` but this does take care of the comments

**Comment by alexandrudima** [comment]
@variab1e When commenting out blocks of code, I personally always choose `//` via `Ctrl+/`. The reason I do it is that `//` can comment out block comments and it is cumulative:
```javascript
	// /**
	//  * Returns the column for the text found at a specific offset inside a rendered dom node
	//  */
	// getColumnOfNodeOffset(lineNumber:number, spanNode:HTMLElement, offset:number): number;

	// // /**
	// //  * Let the line know that decorations might have changed
	// //  */
	// // onModelDecorationsChanged(): void;
```

I am not sure why you choose to comment block of codes with `/**`. Why not use `/*` ?

**Comment by Thaina** [comment]
I think this is annoying behaviour too

**Comment by grolapin** [comment]
Me too. Doesn't seem too hard to fix.

**Comment by briankendall** [comment]
Yup, I am also not thrilled with this would like a way to switch it off.

**Comment by dagit** [comment]
@alexandrudima You asked why someone would do that?

This came up for me recently as I was trying to find some problematic code. Basically doing bisection search commenting things out. I started by commenting out a huge region of code and then incrementally moving the start of the commented region down the file, but the auto inserted characters were a nuisance.

Definitely a +1 from me on making this configurable.

**Comment by ayhanavci** [comment]
For me, this behaviour does more harm than good and is pretty annoying.

**Comment by robotboy655** [comment]
I am also very annoyed by this not being able to be disabled, any features that auto injects text into the editor should have an option to disable it, considering auto closing ```/**/``` comments don't even preserve proper indentation.

**Comment by sunflowr** [comment]
Any news on this? It's really annoying

**Comment by kiznal** [comment]
I see I'm not the only one checking this thread more than three years after it was opened. Any updates on this, MS? Should just be a simple "set this property to false" to not add the extra "*/"...

**Comment by DireLogomachist** [comment]
Adding my voice also here. Incredibly annoying that this cannot be disabled alone.

**Comment by nevercenter** [comment]
Aghghghg this is the worst

**Comment by FereBlanco** [comment]
Try this:

Open the extensions directory
(\Microsoft VS Code\resources\app\extensions\)

There open your languaje directory
(C# for example, \csharp\)

Edit the language configuration file
(language-configuration.json)

From "autoClosingPairs" group delete the line for the comments:
		{ "open": "/*", "close": " */", "notIn": ["string"] }

Save the file. It's worling for me.

**Comment by DireLogomachist** [comment]
@FereBlanco dude you're a lifesaver! This solution works!

**Comment by lonix1** [comment]
@FereBlanco or anyone else - any idea where this can be found in linux?

I looked in `~/.vscode/extensions` but it's not there.

**Comment by TanukiSharp** [comment]
Any news ?

OK we have a good workaround, OK VSCode is entirely free and dudes are making an amazing job, but come on, this kind of details is the very basics of a programmer's like, it shouldn't have to be a hack to tweak this thing :)

**Comment by lonix1** [comment]
@FereBlanco and others - There is one flaw in the workaround, which should be addresses at the same time as the original issue.

When typing `/*`, all folded code below becomes unfolded... what a nuisance!

What I do, when I remember to, is to first type `*/` below the code I want to comment, and then type `/*` above that code block. Then the folding doesn't change.

Has anyone found a workaround for this?

**Comment by Nazg-Gul** [comment]
Unless I'm missing something, the behavior introduced in #73841 can not be undone via configuration.
This is something I find very annoying and rather important to be addressed: this kind of automation is really in a way.

Spent some time this evening getting project compiled and hacked it a little bit. Ended up with [this WIP code](/Nazg-Gul/vscode/tree/wip_auto_closing_comments). It's a so called seems-to-work-for-me state of change. Close to be a proof-of-concept.

Would really be cool if such option is implemented in an official VS Code. I don't mind spending time on wrapping it up, but need input from more experienced VS Code developer about overall design and approach. Not sure if pull-request will be a way to go about it?

**Comment by HDHarry** [comment]
I find this behavior quite annoying too, especially because it currently seems to break automatic generation of doxygen comments via extensions.

**Comment by Nazg-Gul** [comment]
@HDHarry, if it breaks functionality, is probably better to submit a bug report. Such things are very easy to get lost between cracks in the feature-request type of an issue.

At the meantime cleaned up the branch a bit and created #75053.

**Comment by HDHarry** [comment]
Thanks for the hint! I do not have time right now to take care of it, but will do so later.

**Comment by RcSepp** [comment]
Could somebody just fix this already please? It is extremely annoying, having to type "/*" with the key sequence [`/` `*` `DEL` `DEL` `DEL`].

Also, for the future I would recommend implementing the rule that all newly added auto-closing features need to ...
1. be disable-able by configuration.
2. be disabled by default.

After all, in software engineering projects, adding new features that change the behavior in a way that requires adaption from the user (as is clearly the case here) should be avoided if possible and should only happen during mayor releases.

**Comment by PauGuillamon** [comment]
+1, completely agree with all comments here. Clearly this only covers some use cases and breaks others. At least this should be configurable without having to manually edit the language configuration file. As others, I would be very happy if it was disabled by default.

**Comment by MulattoKid** [comment]
Seems to have been fixed in the latest release (1.37.1).

**Comment by lonix1** [comment]
@MulattoKid Please tell us which config option has been added for this. I looked and found nothing.

**Comment by MulattoKid** [comment]
> @MulattoKid Please tell us which config option has been added for this. I looked and found nothing.

So I recently did a fresh VSCode install on Ubuntu 18.04.02. At the time (August 1st), update 1.36.1 was the most recent one. I then immediately updated my settings.json file to be:
```
{
    "workbench.colorTheme": "Visual Studio Light",
    "editor.autoClosingBrackets": "never",
    "editor.autoClosingQuotes": "never",
    "editor.autoIndent": false,
    "editor.acceptSuggestionOnCommitCharacter": false,
    "editor.acceptSuggestionOnEnter": "off",
    "editor.snippetSuggestions": "none",
    "editor.suggest.filterGraceful": false,
    "files.maxMemoryForLargeFilesMB": 8192,
    "editor.wordBasedSuggestions": false
}
```
I quickly noticed that upon entering `/*` in C++ files, the comment was closed on the same line with `*/` (which I had never experienced before - I had an old version before the fresh install).

Then, without changing the settings.json file, I downloaded the newest update, 1.37.1, and the problem disappeared. That's why I thought it had been fixed, but maybe I'm wrong...?

**Comment by aradmidias** [comment]
The issue itself seems to be present (at least for PHP, see below) even in Version `1.78.2`, Commit `b3e4e68a0bc097f0ae7907b217c1119af9e03435`, installed under Linux, Ubuntu as Snap package with the `--classic` option.

> Version: 1.78.2
> Commit: b3e4e68a0bc097f0ae7907b217c1119af9e03435
> Date: 2023-05-10T14:47:05.613Z
> Electron: 22.5.2
> Chromium: 108.0.5359.215
> Node.js: 16.17.1
> V8: 10.8.168.25-electron.0
> OS: Linux x64 6.2.0-20-generic snap
> Sandboxed: Yes

---

The solution, provided by @FereBlanco is not applyable for me; Due to the Snap installation, that File (in my case `/snap/code/current/usr/share/code/resources/app/extensions/php/language-configuration.json` which resolves to `/snap/code/129/usr/share/code/resources/app/extensions/php/language-configuration.json`) is on a read-only; since it is provided by the snap package:
> From "autoClosingPairs" group delete the line for the comments:
> { "open": "/*", "close": " */", "notIn": ["string"] }

So I applied the configuration, suggested by @isidorn and added the following to my User's `settings.json`:
```json
{
	...
	"[php]": {
		...
		"editor.autoClosingBrackets": "never"
	}
	...
}
```

Though this not only disables the autocompletion of `/**`, but also of every Bracket, I type. This behavior fits its name, though it disables way more, than I intended to disable.
I will keep this enabled, since I usually type closing brackets, anyway.

---

@alexdima I can't speak for @variable, but I can explain, how and why this affects me:
I am used to write comments (explaining, why some code exists) using `/**/` comments, so they are clearly distinguishable from code, being commented out (by using `//`).
When doing this, I go used to write `/**/` first and then insert the actual text. This way, writing the comment does not mess up the existing Code, alongside its Syntax-Highlighting, which again prevents the IDE (this is not limited to VSC, but applies to all IDEs) from complaining about Syntax errors, just because I am typing a comment (which I did not close yet).

So whenever I intend to type [`/` `*` `*` `/`], after typing [`/` `*` `*`] the "closing sequence" [`space` `*` `/`] is inserted and I end up with [`/` `*` `*` `/` `space` `*` `/`], which actually does break my code.

Then again, the current autoClosingPair (in PHP, though) would produce a `/** */` comment, which is not, what I want.

---

Here are some things, I would like to note:
1. I abstain from modifying installation files, as the modifications would be lost with the next update.
1. The `language-configuration.json` for `csharp` does not contain the `autoClosingPairs` in question (`{ "open": "/*", "close": " */", "notIn": ["string"] }`) any more (at least not in my installation).
1. The `language-configuration.json` for `php` however **does** contain the following `autoClosingPairs`: `{ "open": "/**", "close": " */", "notIn": [ "string" ] }`
1. **Other languages (`C++` got mentioned, as well) may or may not be affected by this, at time of writing.**

Would it be possible to add an option to **ignore or disable** pre-defined `autoClosingPairs` for a language (or for all languages)?
This would make it easier to disable a specific autoClosingPair, opposed to having this-many Flags to disable this or that.

E.g. (suggestion)
```json
{
	"[php]": {
		"editor.ignoredAutoClosingPairs": [
		    "/*",
		    "/**",
		]
	}
}
```
